Documentation for UOs and some time schemes

GiudGiud opened the pull request.

GiudGiud committed:
Add documentation for verify- and nodal normals user objects

GiudGiud committed:
Delete doc file for deprecated user object

GiudGiud committed:
Add documentation Layered UO, add misc documentation for a time stepper and nodal normals, add some class descriptions refs #18333 #17566 #16248 #15968

GiudGiud committed:
Add docs for Terminator UO
refs #18333 #17566 #16248 #15968

GiudGiud committed:
Add simple documentation for some time first and second order integration schemes, refs #18333 #17566 #16248 #15968

Clang format

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/897094/) on 5e1b367 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/19447/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/19447/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format f16889e13116851ae76fa8f57f2e0ba2b96820be`

GiudGiud committed:
Copy docs in header of diagonal implcit RK time integrating schemes to moosedocs
refs #18333 #17566 #16248 #15968

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/900784/) on ab35365 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/19447/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/900811/) on ab35365 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>bafe7f</th>
      <th colspan=4>#19447 ab3536</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.02%</td>
      <td>82.02%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>72442</td>
      <td>72445</td>
      <td>+3</td>
      <td>5</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15876</td>
      <td>15876</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/19447/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/19447/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/tensor_mechanics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/19447/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented:
need to fix all the butcher tableaus

GiudGiud committed:
Add test and example for terminating-pass a simulation

GiudGiud committed:
Fix formatting for butcher tableau
All tableaus checked except LSDirk 2 which differs from wiki and LSDirk4
Try #5

GiudGiud commented:
struggled with formatting for a long time, but should be ready for review now

joshuahansel commented on test/tests/userobjects/Terminator/terminator_pass.i:
```diff
@@ -0,0 +1,81 @@
+[]
+
+[Variables]
+  [./c]
```
Get these "./" out of here!

joshuahansel commented on test/tests/userobjects/Terminator/terminator_pass.i:
```diff
@@ -0,0 +1,81 @@
+# each solve and can terminate the solve early due to
+# user-defined criteria. (Type: GeneralUserObject)
+#
+# @Requirement F6.40
```
What is this? Is this something we still do or copy-paste error?

joshuahansel commented on framework/doc/content/source/timeintegrators/AStableDirk4.md:
```diff
@@ -5,28 +5,30 @@
+| $b_{j}$ | $\dfrac{1}{24(1/2-\gamma)}$    | $1 - \dfrac{1}{12(1/2-\gamma)^2}$ | $\dfrac{1}{24(1/2-\gamma)^2}$ |
 
-where $\gamma = 1/2 + sqrt(3)/3 * cos(pi/18) ~ 1.06857902130162881$
+where $\gamma = 1/2 + \dfrac{\sqrt(3)}{3} \cos( \dfrac{\pi}{18}) \approx 1.06857902130162881$
```
`\sqrt{3}`

joshuahansel commented on framework/doc/content/source/timeintegrators/LStableDirk3.md:
```diff
@@ -5,25 +5,25 @@
 
-where $\gamma = -sqrt(2)*cos(atan(sqrt(2)/4)/3)/2 + sqrt(6)*sin(atan(sqrt(2)/4)/3)/2 + 1  ~
-.435866521508459$
+where $\gamma = -\sqrt(2) \cos(atan(\sqrt(2)/4)/3)/2 + \sqrt(6) \sin(atan(\sqrt(2)/4)/3)/2 + 1  \approx 0.435866521508459$
```
\sqrt{}

joshuahansel commented on framework/doc/content/source/timeintegrators/Ralston.md:
```diff
@@ -3,8 +3,8 @@
-method and a special case of the 2nd order Runge-Kutta method. It's obtained through an error minimization
-process and has been shown to outperform other 2nd order explicit Runge Kunta methods [!cite](ralston1962).
+method and a special case of the 2nd order Runge-Kutta method. It is obtained through an error minimization
+process and has been shown to outperform other 2nd order explicit Runge Kunta methods, see [!cite](ralston1962).
```
- 2nd-order
- Runge-Kutta
- ", see" -> "; see" (it's a run-on)

joshuahansel reviewed (changes requested):
Mainly commas, hyphens, and formatting issues, some just preference. I think we need to do something else for the Butcher tableau. Having "b_j" in the "c_i" column is confusing. How about we do it with a LaTeX equation so that we can render it in a more traditional manner?

Also (and mainly this is only personal preference) it's easier for me to review if when you fix something in earlier commits, you squash those commits together. This is because I review commit by commit, and sometimes you fix something I've already commented on. Maybe I just need to review the entire PR as one...

GiudGiud commented:
Tableau in equation is genius. I ll do that

GiudGiud committed:
Address Josh s review
Add Butcher tables in equation environment

GiudGiud commented on framework/doc/content/source/userobject/LayeredAverage.md:
```diff
@@ -1,33 +1,68 @@
 
-!syntax description /UserObjects/LayeredAverage
+The layer definition is handled by the `LayeredBase` class. It is summarized here.
+One or multiple layers are defined by their direction (normal or axis), their number
```
ok let me try guessing what you want here lol

joshuahansel replied:
I want a delicious comma after "number"!

GiudGiud commented on framework/doc/content/source/timeintegrators/CrankNicolson.md:
```diff
@@ -1,33 +1,22 @@
 !syntax description /Executioner/TimeIntegrator/CrankNicolson
 
-## Overview
+The Crank Nicholson time integration scheme is a second order implicit method. It is unconditionally stable.
```
he s actually a relative of Jack Nicholson

GiudGiud commented on framework/doc/content/source/timeintegrators/ExplicitEuler.md:
````diff
@@ -1,33 +1,36 @@
-!syntax children /Executioner/TimeIntegrator/ExplicitEuler
-```
-!alert-end!
+The Butcher tableau of the quadrature weights for this method is:
````
well it s easily google-able. I m trying to not loose documentation by doing what I m doing, but also not compete too hard with wikipedia

joshuahansel replied:
That's fine, so I suggest saying something like "This method can be expressed as a Runge-Kutta method with the following Butcher Tableau" (no link necessary).

GiudGiud replied:
ok changed

GiudGiud commented:
Should be good once it passes tests

joshuahansel commented on framework/doc/content/source/timeintegrators/AStableDirk4.md:
````diff
@@ -1,33 +1,45 @@
-```
-!alert-end!
+## Notes:
+- Method is originally due to [!cite](crouzeix1975)
````
Needs blank line before

joshuahansel replied:
And delete the blank line after this item.

joshuahansel replied:
Also it's weird to have a colon after a subsection name

GiudGiud replied:
fixed

joshuahansel commented on framework/doc/content/source/timeintegrators/Ralston.md:
```diff
@@ -1,33 +1,50 @@
+This method can be expressed as a Runge-Kutta method with the following Butcher Tableau:
 
-!syntax inputs /Executioner/TimeIntegrator/Ralston
+!table
```
You forgot this Butcher tableau.

GiudGiud replied:
my b
